main.py

```diff
@@ -1,96 +1,205 @@
 import argparse
-import glob
 import os
-import re
 import time
 
 import tomli
 from tqdm import tqdm
 
 from src.best_blast import find_best_blast
-from src.match_enzyme import run_match_enzyme
+from src.match_enzyme import _run_match_enzyme
 from src.parse_ncbi_xml import parse_blast
 from src.run_scripts import cpu_num, run_blast, run_prodigal
-from src.util import make_dir
+from src.util import make_dir, find_file
 
 
-def main():
-    time_start = time.time()
-    abs_dir = os.path.dirname(__file__)
+ROOT_DIR = os.path.dirname(__file__)
+PRODIGAL_FOLDER = os.path.join(ROOT_DIR, "tmp/prodigal")
+BLAST_FOLDER = os.path.join(ROOT_DIR, "tmp/blast")
+PARSE_BLAST_FOLDER = os.path.join(ROOT_DIR, "tmp/parse_blast")
+BEST_BLAST_FOLDER = os.path.join(ROOT_DIR, "tmp/best_blast")
+ENZYME_MAPPING_FOLDER = os.path.join(ROOT_DIR, "result")
+
+# utils
+def load_config():
+    with open(os.path.join(ROOT_DIR, "config.toml"), "rb") as f:
+        config = tomli.load(f)
+    return config
+
+def check_blast_database(database_path):
+    print("Check blast database existence")
+    if os.path.isfile(database_path):
+        print(f"Database: {os.path.basename(database_path)}")
+    else:
+        raise Exception("Blast database does not exist. Check config.toml before running.")
+
+def run_and_save(func, output_path, file_list, file_save_format, **kwargs):
+    for filename in tqdm(file_list):
+        basename_old_extension = os.path.basename(filename)
+        basename = basename_old_extension.rsplit(".", 1)[0]
+        basename_new_extension = f"{basename}.{file_save_format}"
+        output_name = os.path.join(output_path, basename_new_extension)
+        if kwargs is not None:
+            func(filename, output_name, **kwargs)        
+        else:
+            func(filename, output_name)
```

Comment:
By the way, in line 42:
Is it the suggested way to extract kwargs by using ** and pass it to other functions?

Reply:
`**kwrags` works well, but I think the run_and_save() is making things less transparent.